javaslang/src-gen/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,437 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+/*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*\
+   G E N E R A T O R   C R A F T E D
+\*-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-*/
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import javaslang.collection.List;
+import javaslang.concurrent.Future;
+
+import javaslang.Function3;
+import javaslang.Function4;
+import javaslang.Function5;
+import javaslang.Function6;
+import javaslang.Function7;
+import javaslang.Function8;
+import javaslang.control.Option;
+import javaslang.control.Try;
+import javaslang.concurrent.Future;
+
+/**
+ * Applicative helpers
+ *
+ * @author Daniel Dietrich
+ * @since 2.1.0
+ */
+public final class Applicative {
```

Comment:
I think I do not understand Applicative and Monad right.

For me, Monad is just a type having map() and flatMap(), obeying specific laws. In our lift() implementations we make use of map() and (!) flatMap(). Because we use flatMap() and Option et al. are monads, this actually _is_ monadic lifting.

Applicative has a sequence operator, in Haskell `<*>`. Our types do not have such an operator. Also flatMap() does not allow us do go down the chain once empty, let's say a None, is reached:

```java
opt1.flatMap(o1 -> 
opt2.flatMap(o2 -> // <=== We never get here, if opt1 is None.
opt3.map(o3 -> f.apply(o1, o2, o3));
```